crates/ra_ide_api/src/call_info.rs

```diff
@@ -49,6 +58,7 @@ pub(crate) fn call_info(db: &RootDatabase, position: FilePosition) -> Option<Cal
                 .args()
                 .position(|arg| arg.syntax().range().contains(position.offset))
                 .or(Some(num_params - 1))
+                .min(Some(num_args_of_call))
```

Comment:
Hm, it's unclear why this logic is correct... Perhaps something like `args.take_while(|it| it.range() < position.offset).count()` would work?

Reply:
It's basically doing the `min` of the function signature's arguments versus the call site's. I'll see if I can make it more obvious.

Reply:
Thanks that cleaned it up quite a bit.